.github/workflows/update-branch-version.yml

```diff
@@ -0,0 +1,148 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "Release: 0. Update version in target branch"
+
+# The target branch when starting this workflow should be:
+# 1. "branch/{major}.{minor}.x" if it exists, or
+# 2. "main"
+
+on:
+  workflow_dispatch:
+    inputs:
+      new_version:
+        description: "Version 'X.Y.Z' for the release branch."
+        type: string
+        required: true
+        default: "0.0.0"
+      target_branch:
+        description: "Target branch for the version update"
+        type: string
+        required: false
+        default: "main"
+      force:
+        description: "Enable overwriting existing PR branches (this does not force overwrite the target branch or skip creating a PR)"
+        type: boolean
+        required: true
+        default: false
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -euo pipefail {0}
+
+jobs:
+  update-version:
+    permissions:
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.target_branch }}
+
+      - name: Verify run from main
+        id: verify-main
+        run: |
+          # This action can only be run from main.
+          if [[ $GITHUB_REF != refs/heads/main ]]; then
+            echo "  This action may only be run fom main" | tee -a $GITHUB_STEP_SUMMARY
+            exit 1
+          fi
+
+      - name: Prepare environment
+        id: prepare-env
+        run: |
+          log_and_export_vars() {
+            for var in "$@"; do
+              printf "%-15s %s\n" "$var:" "${!var}" | tee -a $GITHUB_STEP_SUMMARY
+              echo "${var}=${!var}" | tee -a $GITHUB_ENV | tee -a $GITHUB_OUTPUT
+            done
+          }
+
+          full_version=${{ inputs.new_version }}
+          major_version=$(echo ${full_version} | cut -d. -f1)
+          minor_version=$(echo ${full_version} | cut -d. -f2)
+          patch_version=$(echo ${full_version} | cut -d. -f3)
+          branch_name=${{ inputs.target_branch }}
```

Comment:
suggestion: Should we sanity check that the branch name agrees with the major/minor version? e.g., so that if I run this action to update `branch/2.3.x` to version `2.4.0` it would fail.

Reply:
Should we? This ultimately just emits a PR which we can vote down or close.

Reply:
Action now fails when given a branch target and the versions mismatch.

https://github.com/wmaxey/cccl/actions/runs/11390484752